infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/SinkWriter.java

```diff
@@ -29,12 +31,18 @@ public abstract class SinkWriter {
     // Registry table for verifying replicated entries from Source side.
     final CorfuTable<TableName, CorfuRecord<TableDescriptors, TableMetadata>> registryTable;
 
+    // Configuration for LR in Source / Sink cluster.
+    final LogReplicationConfig config;
+
     // Limit the initialization of this class only to its children classes.
-    SinkWriter(CorfuRuntime rt) {
+    SinkWriter(CorfuRuntime rt, LogReplicationConfig config) {
         this.rt = rt;
         // Instantiate TableRegistry and register serializers in advance to avoid opening
         // registry table in log entry writer's transaction
         this.registryTable = rt.getTableRegistry().getRegistryTable();
+        this.config = config;
+        // Signal the config to sync with registry table during initialization
+        config.syncWithRegistry();
```

Comment:
A sync is not needed here.  At the time of initialization, the config is passed to the writers.  It is then performed on entry to each state(snapshot and log entry).  Later, it is done on-the-fly as new streams are received.